sdks/universal-router-sdk/src/swapRouter.ts

```diff
@@ -162,6 +162,28 @@ export abstract class SwapRouter {
       planner.addCommand(CommandType.V3_POSITION_MANAGER_CALL, [v3Call])
     }
 
+    // if migrate options has a currency, require a batch permit
+    if (options.v4AddLiquidityOptions.migrateOptions?.neededCurrency) {
+      // need to permit the UR to spend your currency on permit2
+      if (options.v4AddLiquidityOptions.batchPermit) {
+        invariant(
+          options.v4AddLiquidityOptions.batchPermit.permitBatch.spender == universalRouterAddress,
+          'INVALID_SPENDER'
+        )
+        planner.addCommand(CommandType.PERMIT2_PERMIT_BATCH, [
```

Comment:
its only ever going to be 1 currency i think? so using `batch` is unnecessary?

Reply:
permitBatch is already a param of v4AddLiquidityOptions so i figured i would just keep it instead of adding another one?